File: in_Python/0345 Reverse Vowels of a String.py
class Solution(object):
    def isV(self,a):
        return a in ['a','e','i','o','u','A','E','I','O','U']
    
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        i = 0
        j = len(s)-1
        
        ss = []
        for x in s: ss.append(x)
        
        while i < j:
            while not self.isV(s[i]) and i < j:
                i += 1
            while not self.isV(s[j]) and i < j:
                j -= 1
            t = ss[i]
            ss[i] = ss[j]
            ss[j] = t
            i += 1
            j -= 1
            
        res = ""
        for x in ss: res += x
        return res
    

File: in_Java/0345 Reverse Vowels of a String.java
class Solution {
    public String reverseVowels(String s) {
        char[] ch = s.toCharArray();
        int i = 0; int j = ch.length-1;
        char t;
        while(i < j){
            while(!isV(ch[i]) && i < j) i++;
            while(!isV(ch[j]) && i < j) j--;
            t = ch[i];
            ch[i] = ch[j];
            ch[j] = t;
            i++;
            j--;
        }
        return String.valueOf(ch);
    }
    private boolean isV(char c){
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') return true;
        else if(c == 'A' || c == 'E' || c == 'I' || c == 'O' || c == 'U') return true;
        else return false;
    }
}